srcStatic/GameModules/DllModule.cpp

```diff
@@ -18,4 +19,56 @@ void DllModule::LoadModuleDll(const std::string& dllPath)
 	}
 
 	this->moduleDllHandle = dllHandle;
+
+	DetectExportedModuleFunctions();
+}
+
+void DllModule::DetectExportedModuleFunctions()
+{
+	loadModuleFunctionIni = (LoadModuleFunctionIni)GetProcAddress(moduleDllHandle, "InitMod");
+	if (loadModuleFunctionIni == nullptr) {
+		loadModuleFunctionConsole = (LoadModuleFunctionConsole)GetProcAddress(moduleDllHandle, "mod_init");
+	}
+
+	unloadModuleFunction = (UnloadModuleFunction)GetProcAddress(moduleDllHandle, "DestroyMod");
+	if (unloadModuleFunction == nullptr) {
+		unloadModuleFunction = (UnloadModuleFunction)GetProcAddress(moduleDllHandle, "mod_destroy");
+	}
+
+	runModuleFunction = (RunModuleFunction)GetProcAddress(moduleDllHandle, "RunMod");
+	if (runModuleFunction == nullptr) {
+		runModuleFunction = (RunModuleFunction)GetProcAddress(moduleDllHandle, "mod_run");
+	}
+}
+
+void DllModule::Load()
+{
+	if (loadModuleFunctionIni != nullptr) {
+		loadModuleFunctionIni(Name().c_str());
+	}
+	else if (loadModuleFunctionConsole != nullptr) {
+		loadModuleFunctionConsole();
+	}
+}
+
+bool DllModule::Unload()
+{
+	bool success = true;
+
+	if (unloadModuleFunction != nullptr) {
+		success = unloadModuleFunction();
```

Comment:
Not all modules return a success value during unload.

We should probably log unload errors. (Console modules only).